src/main/java/mSearch/tool/Functions.java

```diff
@@ -157,23 +157,35 @@ public static String getCompileDate() {
         return msg;
     }
 
+    public static Version getProgVersion() {
+        final ResourceBundle rb;
+        String TOKEN_VERSION = "VERSION";
+        try {
+            ResourceBundle.clearCache();
+            rb = ResourceBundle.getBundle("version");
+            if (rb.containsKey(TOKEN_VERSION)) {
+                return new Version(rb.getString(TOKEN_VERSION));
+            }
+        } catch (Exception e) {
+            Log.errorLog(134679898, e);
+        }
+        return new Version("");
+    }
+    
+    @Deprecated
     public static String getBuildNr() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Add the missing @deprecated Javadoc tag. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1133)